packages/ra-core/src/controller/RecordContext.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { createContext, useContext, ReactNode } from 'react';
+import pick from 'lodash/pick';
+import { Record } from '../types';
+
+/**
+ * Context to store the result of the useRecord() hook.
+ *
+ * Use the useRecordContext() hook to read the context. That's what the Edit and Show components do in react-admn.
+ *
+ * @example
+ *
+ * import { useEditController, EditContext } from 'ra-core';
+ *
+ * const Edit = props => {
+ *     const controllerProps = useEditController(props);
+ *     return (
+ *         <RecordContextProvider value={controllerProps}>
+ *             ...
+ *         </RecordContextProvider>
+ *     );
+ * };
+ */
+export const RecordContext = createContext<Record | Omit<Record, 'id'>>(
+    undefined
```

Comment:
no need for undefined, just pass no value

Reply:
TS disagree